biz.aQute.bndlib/src/aQute/bnd/exporter/runbundles/RunbundlesExporter.java

```diff
@@ -33,23 +36,35 @@ public RunbundlesExporter() {}
 	public Entry<String, Resource> export(String type, Project project, Map<String, String> options) throws Exception {
 		project.prepare();
 		Collection<Container> runbundles = project.getRunbundles();
-		Jar jar = new Jar(project.getName());
+		String prefix = options.getOrDefault(TARGET_DIR, "");
+		if (!prefix.endsWith("/")) {
+			prefix = prefix.concat("/");
+		}
+		String template = options.get(TEMPLATE);
+
+		Jar jar = null;
+		if (template == null) {
+			jar = new Jar(project.getName());
+		} else {
+			jar = new Jar(project.getName(), template);
+		}
 		jar.setDoNotTouchManifest();
 		for (Container container : runbundles) {
 			File source = container.getFile();
-			String path = nonCollidingPath(jar, source.getName());
+			String path = nonCollidingPath(jar, source.getName(), prefix);
 			jar.putResource(path, new FileResource(source));
 		}
 		return new SimpleEntry<>(jar.getName(), new JarResource(jar, true));
 	}
 
-	private String nonCollidingPath(Jar jar, String fileName) {
+	private String nonCollidingPath(Jar jar, String fileName, String prefix) {
 		String[] parts = Strings.extension(fileName);
 		if (parts == null) {
 			parts = new String[] {
 				fileName, ""
 			};
 		}
+		fileName = prefix.concat(fileName);
```

Comment:
If you enter the while loop, the prefix is lost. You should also add a test for this.

Reply:
Good point. I totally missed that part. done